tests/functional/object/mcg/test_internal_route_auth.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import requests
+
+from ocs_ci.ocs import constants
+from ocs_ci.helpers.helpers import get_noobaa_metrics_token_from_secret
+from ocs_ci.ocs.resources.pod import get_noobaa_endpoint_pods
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework.pytest_customization.marks import (
+    tier2,
+    red_squad,
+    mcg,
+    post_upgrade,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@mcg
+@red_squad
+class TestNoobaaRouteAuthentication:
+    """
+    Tests noobaa management route and noobaa endpoints with JWT authentication
+    """
+
+    @post_upgrade
```

Comment:
Will we still need to include this test in upgrade runs of versions >= 4.21 where the feature will exist both pre- and post-upgrade?

If not, please create an issue to remove the `post_upgrade` decorator in the next version so we don’t run the test unnecessarily often.

Reply:
https://github.com/red-hat-storage/ocs-ci/issues/12974